### PR TITLE
feat: Config diff 3 oss

### PR DESCRIPTION
**Pull request by prakash100198**
# Description

Fixes #

## Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: api/router/DeploymentConfigRouter.go [hunk 1]
```diff
@@ -26,6 +26,6 @@ func (router DeploymentConfigurationRouterImpl) initDeploymentConfigurationRoute
 		Methods("GET")
 	configRouter.Path("/data").
 		HandlerFunc(router.deploymentGroupRestHandler.GetConfigData).
-		Methods("GET")
+		Methods("POST")
```
Comment (vikramdevtron) [inline api/router/DeploymentConfigRouter.go:29]: backward compatibility ? new orchestrator build with old dashboard build

Comment (prakash100198) [reply api/router/DeploymentConfigRouter.go]: done

**Review by gireesh-naidu — APPROVED** [review]

**Review by gireesh-naidu — APPROVED** [review]

File: pkg/generateManifest/DeploymentTemplateService.go [hunk 2]
```diff
@@ -217,16 +220,54 @@ func (impl DeploymentTemplateServiceImpl) GetDeploymentTemplate(ctx context.Cont
...
+		result.VariableSnapshot = variableSnapshot
+		if response != nil {
+			result.Data = response.Data
+			result.ResolvedData = response.ResolvedData
+			result.VariableSnapshot = response.VariableSnapshot
+			result.TemplateVersion = response.TemplateVersion
+			result.IsAppMetricsEnabled = response.IsAppMetricsEnabled
+		}
+		return result, nil
+	}
```
Comment (kartik-579) [inline pkg/generateManifest/DeploymentTemplateService.go:245]: move with other adapters

File: pkg/generateManifest/DeploymentTemplateService.go [hunk 3]
```diff
@@ -217,16 +220,54 @@ func (impl DeploymentTemplateServiceImpl) GetDeploymentTemplate(ctx context.Cont
...
+	if variableSnapshot != nil {
+		result.VariableSnapshot = variableSnapshot
+	}
+	request = impl.setRequestMetadata(&request)
+	manifest, err := impl.GenerateManifest(ctx, &request, resolvedValue)
+	if err != nil {
+		return result, err
+	}
+	result.Data = *manifest.Manifest
```
Comment (kartik-579) [inline pkg/generateManifest/DeploymentTemplateService.go:254]: manifest nil check

File: pkg/pipeline/history/ConfigMapHistoryService.go [hunk 4]
```diff
@@ -691,3 +698,129 @@ func (impl ConfigMapHistoryServiceImpl) CheckIfTriggerHistoryExistsForPipelineId
...
+	resolvedConfigDataString, err := utils.ConvertToString(resolvedConfigDataReq)
+	if err != nil {
+		impl.logger.Errorw("getCmCsPublishedConfigResponse, error in converting config data to json raw message", "pipelineId", pipelineId, "deployedOn", deployedOn, "err", err)
+		return nil, err
+	}
+	resolvedConfigDataStringJson, err := utils.ConvertToJsonRawMessage(resolvedConfigDataString)
+	if err != nil {
+		impl.logger.Errorw("getCmCsPublishedConfigResponse, error in ConvertToJsonRawMessage for resolvedJson", "resolvedJson", resolvedConfigDataStringJson, "err", err)
+		return nil, err
+	}
```
Comment (kartik-579) [inline pkg/pipeline/history/ConfigMapHistoryService.go:775]: same marshal unmarshal flow as before

**Review by kartik-579 — CHANGES_REQUESTED** [review]

File: pkg/configDiff/DeploymentConfigurationService.go [hunk 5]
```diff
@@ -82,30 +125,367 @@ func (impl *DeploymentConfigurationServiceImpl) ConfigAutoComplete(appId int, en
...
+		WithPipelineStrategyMetadata(pipelineStrategyHistory.PipelineTriggerType, string(pipelineStrategyHistory.Strategy))
+	return pipelineConfig, nil
+}
+
+func (impl *DeploymentConfigurationServiceImpl) getDeploymentsConfigForPreviousDeployments(ctx context.Context, configDataQueryParams *bean2.ConfigDataQueryParams,
+	appId, envId int) (generateManifest.DeploymentTemplateResponse, error) {
+	deploymentTemplateRequest := generateManifest.DeploymentTemplateRequest{
+		PipelineId:                  configDataQueryParams.PipelineId,
+		DeploymentTemplateHistoryId: configDataQueryParams.IdentifierId,
+		RequestDataMode:             generateManifest.Values,
```
[the remark this answers is not in the document]
Comment (prakash100198) [reply pkg/configDiff/DeploymentConfigurationService.go:420]: no this is correct, had to pass Values or else the function where this is being passed will give manifest as response.

**Review by kartik-579 — APPROVED** [review]

**Review by nishant-d — APPROVED** [review]

**Pull request merged**